### DISCUSSION TITLE
Interaction with some parts of complex model

### DISCUSSION THREAD

**Discussion by Ncit** [sceneview/sceneform-android#395] [Q&A] [answered]
Hello everyone,
Is it possible to glow some parts of gltf model and add touchListener on it? Because I need to open dialog screens when user interacts with them, but did not find any solution for now
<img width="426" alt="180644034-c9ffd4c9-2993-4e38-a2df-250f1e081d80" src="https://user-images.githubusercontent.com/2179841/181076385-4e90a047-7fd0-48b1-9a78-821cbbd6bce0.png">
.

**Comment by grassydragon** [comment, marked as answer]
Hi!
Yes, detecting when a part of the model is tapped is supported in [SceneView](https://github.com/SceneView/sceneview-android). However, the parts of the model should be highlighted either by changing a material or based on animations in glTF.

**Reply by Ncit** [reply]
Is it possible to point me on functions, which used to detect tap? Also I did not get about materials, can you please explain what it is?

**Reply by grassydragon** [reply]
You can use this callback for all nodes: https://github.com/SceneView/sceneview-android/blob/848a95831d895a9c61143dc042195632f5505199/sceneview/src/main/java/io/github/sceneview/SceneView.kt#L310
Or set a callback for a particular node: https://github.com/SceneView/sceneview-android/blob/848a95831d895a9c61143dc042195632f5505199/sceneview/src/main/java/io/github/sceneview/node/Node.kt#L401
Speaking about materials, I mean that SceneView doesn't highlight the model parts on its own and you should find a way to do so.

**Reply by Ncit** [reply]
Thanks, interaction works. But issue with that renderable id always changes((

**Reply by grassydragon** [reply]
However, if you find nodes by name in the model using Filament, you should be able to map the renderable ID to the node name.

**Comment by Ncit** [comment]
thanks for help!!
